src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/PhysicalFileProviderTests.netcoreapp.cs

```diff
@@ -0,0 +1,146 @@
+            File.CreateSymbolicLink(linkPath, filePath);
+
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(useWildcard ? "*" : linkName);
+            Assert.False(token.HasChanged);
+
+            // Act
+            File.WriteAllText(filePath, "v1.2");
+            Thread.Sleep(GetTokenPollingInterval(token));
+
+            // Assert
+            Assert.True(token.HasChanged);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void UsePollingFileWatcher_UseActivePolling_HasChanged_SymbolicLink_TargetNotExists(bool useWildcard)
+        {
+            // Arrange
+            using var rootOfLink = new DisposableFileSystem();
+            string linkName = Path.GetRandomFileName();
+            string linkPath = Path.Combine(rootOfLink.RootPath, linkName);
+            File.CreateSymbolicLink(linkPath, "not-existent-file");
+
+            // Act
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(useWildcard ? "*" : linkName);
+
+            // Assert
+            Assert.False(token.HasChanged);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void UsePollingFileWatcher_UseActivePolling_HasChanged_SymbolicLink_TargetChanged(bool useWildcard)
+        {
+            // Arrange
+            using var rootOfFile = new DisposableFileSystem();
+            string file1Path = Path.Combine(rootOfFile.RootPath, Path.GetRandomFileName());
+            File.WriteAllText(file1Path, "v1.1");
+
+            string file2Path = Path.Combine(rootOfFile.RootPath, Path.GetRandomFileName());
+            File.WriteAllText(file2Path, "v2.1");
+
+            using var rootOfLink = new DisposableFileSystem();
+            string linkName = Path.GetRandomFileName();
+            string linkPath = Path.Combine(rootOfLink.RootPath, linkName);
+            File.CreateSymbolicLink(linkPath, file1Path);
+
+            string filter = useWildcard ? "*" : linkName;
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(filter);
+            Assert.False(token.HasChanged);
+
+            // Act 1 - Change file 1's content.
+            File.WriteAllText(file1Path, "v1.2");
+            Thread.Sleep(GetTokenPollingInterval(token));
+
+            // Assert 1
+            Assert.True(token.HasChanged);
+
+            // Act 2 - Change link target to file 2.
+            token = provider.Watch(filter); // Once HasChanged is true, the value will always be true. Get a new change token.
+            Assert.False(token.HasChanged);
+            File.Delete(linkPath);
```

Comment:
Through `File.CreateSymbolicLink`, it is not. It may be possible to modify the target of an existing link by using `FSCTL_SET_REPARSE_POINT`, see https://docs.microsoft.com/en-us/windows/win32/fileio/reparse-point-operations. 

For Unix, I know that you can use the -f switch in  `ln -sf bar foo` to force the link creating even if it already exists, but I guess that's just deleting the old link and creating it again.

Reply:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/ln.html says that `-f` in `ln` is functionally equivalent to "call unlink", which we call `File.Delete`.